src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -773,15 +775,44 @@ double dimensionful_spin_magnitude(
 }
 
 template <typename Frame>
-std::array<double, 3> spin_vector(const double spin_magnitude,
-                                  const Scalar<DataVector>& area_element,
-                                  const Scalar<DataVector>& radius,
-                                  const tnsr::i<DataVector, 3, Frame>& r_hat,
-                                  const Scalar<DataVector>& ricci_scalar,
-                                  const Scalar<DataVector>& spin_function,
-                                  const YlmSpherepack& ylm) noexcept {
-  std::array<double, 3> spin_vector = {{0.0, 0.0, 0.0}};
+void spin_vector(const gsl::not_null<std::array<double, 3>*> result,
+                 const double spin_magnitude,
+                 const Scalar<DataVector>& area_element,
+                 const Scalar<DataVector>& radius,
+                 const tnsr::i<DataVector, 3, Frame>& r_hat,
+                 const Scalar<DataVector>& ricci_scalar,
+                 const Scalar<DataVector>& spin_function,
+                 const Strahlkorper<Frame>& strahlkorper) noexcept {
+  const auto& ylm = strahlkorper.ylm_spherepack();
+  // we want to assert that the DataVectors in area_element, radius,
+  // r_hat, ricci_scalar, and r_hat have the same size as the ylm size
+
+  // get the ylm's physical size as a variable to reuse
+  const size_t ylm_physical_size = ylm.physical_size();
+  ASSERT(get(area_element).size() == ylm_physical_size,
+         "area_element size doesn't match ylm physical size: "
+             << get(area_element).size() << " vs " << ylm_physical_size);
+  ASSERT(get(radius).size() == ylm_physical_size,
+         "radius size doesn't match ylm physical size: "
+             << get(radius).size() << " vs " << ylm_physical_size);
+  ASSERT(r_hat.get(0).size() == ylm_physical_size and
```

Comment:
`get<0>(r_hat)` instead of r_hat.get(0). When you know the component you want to get, spectre provides a compile-time version of get, which will make runtime faster. It only works when you know the component you want to get at compile time (template parameters must be known at compile time, not at runtime)